### DISCUSSION TITLE
These credentials do not match our records.

### DISCUSSION THREAD

**Discussion by Makemaks** [filamentphp/filament#11506] [Help]
### Package

Other

### Package Version

^3.2

### How can we help you?

I tried logging in with correct credentials and this is what I encounter. My friend is using windows and we pulled from master and did all the composer and npm without an error and imported the same database. He was able to login but in my case I got 'These credentials do not match our records.'. I even try to change it's password but still got the same error. I use this seeder to change password and I double check if the password is really password123 and still have the same error 'These credentials do not match our records.'
![image](https://github.com/filamentphp/filament/assets/93358735/c7bd8f52-9ef6-4dd3-8a00-300dccd3364e)
![image](https://github.com/filamentphp/filament/assets/93358735/6fa98b9c-399e-4f23-9161-e383e048a2ec)
<img width="1440" alt="image" src="https://github.com/filamentphp/filament/assets/93358735/522faaea-3ba7-4a88-acb6-e42787faecca">
<img width="1195" alt="image" src="https://github.com/filamentphp/filament/assets/93358735/f9f31cc6-c863-45a7-bde1-d07b10fd0488">

**Comment by leandrocfe** [comment]
Did you try to create another user? 
`php artisan make:filament-user`

**Comment by goyne** [comment]
Could it be that **email_verified_at** is empty?

I had the same error when this field was empty. Laravel normally redirects the user to the verify email page, but for some reason Filament doesn't seem to do that.

**Reply by leandrocfe** [reply]
If you are using [MustVerifyEmail](https://laravel.com/docs/10.x/verification#model-preparation) I guess.

**Reply by goyne** [reply]
Yes and no 🙃
With and without MustVerifyEmail, that error is given.

But I have just found that if I don't implement the FilamentUser contract that everything works as expected. And that's quite strange. Seems then that the configuration of the panel is not quite right yet?

See also #11679

**Comment by SlimDeluxe** [comment]
The error message can be misleading.
It seems Filament uses the "canAccessPanel" method on the login attempt, so any restrictions you have specified there must be met for the seeded user if you want to login with that one.

**Reply by ryanmillergm** [reply]
@SlimDeluxe Yeah. I had the same issue. For me it was just updating the canAccessPanel function in User model. I removed the email validation '@yourdomain.com'

```
public function canAccessPanel(Panel $panel): bool
    {
        return str_ends_with($this->email, '@yourdomain.com') && $this->hasVerifiedEmail();
    }
```

**Reply by edjw** [reply]
In case it's helpful, I changed my `canAccessPanel` function in `User.php` to this to let me allow only specific domains and also specific emails

```php
public function canAccessPanel(Panel $panel): bool
    {

        // Add a comma separated list of valid domains to the VALID_USER_DOMAINS environment variable.

        // Or add a comma separated list of valid emails to the VALID_USER_EMAILS environment variable.

        $valid_user_domains = explode(',', env('VALID_USER_DOMAINS'));

        // $valid_user_emails = explode(',', env('VALID_USER_EMAILS'));

        foreach ($valid_user_domains as $domain) {
            if (str_ends_with($this->email, $domain)) {
                return true;
            }
        }

        // foreach ($valid_user_emails as $email) {
        //     if ($this->email == $email) {
        //         return true;
        //     }
        // }

        return false;
    }
```

**Comment by hyphen81** [comment]
I'm having this same issue on my production environment on an app I rolled out last week. I have 3 panels, 2 of which are internal only, and one is client facing. All my internal folks can login just fine, but clients can't. I have the `MustVerifyEmail` trait and I'm implementing the `FilamentUser` contract. This is my `canAccessPanel` method:

```
    public function canAccessPanel(Panel $panel): bool
    {
        if ($panel->getId() == 'client') {
            return $this->hasVerifiedEmail();
        } else if ($panel->getId() == 'admin') {
            return $this->role->is_internal && $this->hasVerifiedEmail() && $this->hasAdminRole();
        } else if ($panel->getId() == 'app') {
            return $this->role->is_internal && $this->hasVerifiedEmail();
        } else {
            return true;
        }
    }
```

Did anyone figure this out? What makes this more confusing is that I can't reproduce it locally, but it's happening in production. Totally lost on what might be the root cause.

**Reply by hyphen81** [reply]
In my case, it doesn't even seem to be getting to the canAccessPanel method before failing.
